### ISSUE TITLE
[Doc]:  vllm的beam-search执行逻辑

### ISSUE THREAD

**Issue by qlibp** [qlibp/vllm-ascend#20]
### 📚 The doc issue

vllm中的beam-search概念和需要和reqloop对齐，不然进行benchmark时会有问题

**Comment by qlibp** [comment]
beam_width: 保留多少条候选序列
n（API 层）: OpenAI API 中 n 映射为 beam_width

**Comment by qlibp** [comment]
每个 beam 在每步都是一个独立的 engine request，有独立的 request_id（如 reqid-beam-0, reqid-beam-1）。调度器、KV cache 管理器对 beam search 完全无感知
   - 劣势：beam 之间 KV cache 无法共享，内存开销大

**Comment by qlibp** [comment]
OpenAI Compatible参数的`use_beam_search`, `best_of`, `n`, `top_k`之间的关系

功能描述：
- n: 最终返回几条完整的生成结果
- best_of: 服务端内部实际生成多少条候选
- top_k: 采样时只从概率最高的 k 个 token 里选下一个 token (top-k中随机选1个)
- use_beam_search: 每一步按累计概率扩展(累计概率)

互斥关系：
- use_beam_search 与 top_k
- use_beam_search 与 best_of

约束关系：
- best_of >= n
- stream=true，只能n=1

**Comment by qlibp** [comment]
以vllm为例，每个活跃 beam 生成 2W(e.g. 2*beam_width) 个新候选 → 总候选数 = 当前beam数 × 2W
- 2W含义: 取 logprob 为 top-(2×W) 个token

**Comment by qlibp** [comment]
- model-fwd 得到 hidden_states -> hidden_states 经过lm_head得到logits -> logits 通过softmax得到log_prob -> 通过log_prob得到token-id
- 从 logits 到 log_prob 的计算时，logits是一个向量，长度为vocab_size
- log_prob通过softmax算得

**Comment by qlibp** [comment]
Beam Search特性: 包含 Sampler 内的修改(采样出top-2W token) 和 beam search 管理器(负责剪枝)

**Comment by qlibp** [comment]
为什么候选sequence固定取2W而不是W/1.5W/3W?

**Comment by qlibp** [comment]
## vllm offline beam-search 执行流程

## vllm online beam-search 执行流程
